### PR TITLE
VMSS: support for eviction policy

### LINKED ISSUE #1690: azurerm_virtual_machine_scale_set: support eviction policy
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

With low-priority scale sets, it can be interesting to set the 'eviction policy', cf https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-use-low-priority#eviction-policy.

> When creating low-priority scale sets, you can set the eviction policy to Deallocate (default) or Delete.
> 
> The Deallocate policy moves your evicted VMs to the stopped-deallocated state allowing you to redeploy evicted instances. However, there is no guarantee that the allocation will succeed. The deallocated VMs will count against your scale set instance quota and you will be charged for your underlying disks.

### New or Affected Resource(s)

* azurerm_virtual_machine_scale_set

### Potential Terraform Configuration

```hcl
resource "azurerm_virtual_machine_scale_set" "test" {
  name                = "acctvmss-%[1]d"
  location            = "${azurerm_resource_group.test.location}"
  resource_group_name = "${azurerm_resource_group.test.name}"
  upgrade_policy_mode = "Manual"
  overprovision       = false
  priority            = "Low"
  eviction_policy = "Delete"
```

### References

* https://docs.microsoft.com/en-us/rest/api/compute/virtualmachinescalesets/createorupdate#virtualmachineevictionpolicytypes

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Fixes #1690

### CODE REVIEW

File: website/docs/r/virtual_machine_scale_set.html.markdown [hunk 1]
```diff
@@ -269,24 +269,25 @@ The following arguments are supported:
...
 * `os_profile` - (Required) A Virtual Machine OS Profile block as documented below.
 * `os_profile_secrets` - (Optional) A collection of Secret blocks as documented below.
 * `os_profile_windows_config` - (Required, when a windows machine) A Windows config block as documented below.
 * `os_profile_linux_config` - (Required, when a linux machine) A Linux config block as documented below.
 * `network_profile` - (Required) A collection of network profile block as documented below.
+* `overprovision` - (Optional) Specifies whether the virtual machine scale set should be overprovisioned.
+* `rolling_upgrade_policy` - (Optional) A `rolling_upgrade_policy` block as defined below. This is only applicable when the `upgrade_policy_mode` is `Rolling`.
+* `single_placement_group` - (Optional) Specifies whether the scale set is limited to a single placement group with a maximum size of 100 virtual machines. If set to false, managed disks must be used. Default is true. Changing this forces a
+  new resource to be created. See [documentation](http://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-placement-groups) for more information.
```
Comment (katbyte) [inline website/docs/r/virtual_machine_scale_set.html.markdown]: Should there be a line break here 🤔

Comment (tombuildsstuff) [reply website/docs/r/virtual_machine_scale_set.html.markdown]: nope - good spot 👍

Comment (tombuildsstuff) [reply website/docs/r/virtual_machine_scale_set.html.markdown]: (pushed a commit to sort these fields into Required -> Optional & alphabetical order too)

**Comment by tombuildsstuff** [comment]
Tests pass:

screenshot 2018-11-04 at 21 12 02

**Pull request merged**